src/program/config/README.md

```diff
@@ -144,6 +145,70 @@ Likewise, to change the port for `1.2.3.4`, do:
 $ snabb config set ID /routes/route[addr=1.2.3.4]/port 7
 ```
 
+The general rule for paths and value syntax is that if a name appears in
+the path, it won't appear in the value.  Mostly this works as you would
+expect, but there are a couple of edge cases for instances of `list` and
+`leaf-list` nodes.  For example:
+
+```
+$ snabb config get ID /routes/route[addr=1.2.3.4]/port
+1
+$ snabb config get ID /routes/route[addr=1.2.3.4]
+port 1;
+$ snabb config get ID /routes/route
+{
+  addr 1.2.3.4;
+  port 1;
+}
+{
+  addr 2.3.4.5;
+  port 2;
+}
+$ snabb config get ID /routes
+route {
+  addr 1.2.3.4;
+  port 1;
+}
+route {
+  addr 2.3.4.5;
+  port 2;
+}
+```
+
+Note the case when getting the `list` `/routes/route`:  the syntax is a
+sequence of brace-delimited entries.
+
+To select an entry from a `leaf-list`, use the `position()` selector:
+
+```
+$ snabb config get ID /public-ip[position()=1]
+10.10.10.10
+$ snabb config get ID /public-ip[position()=2]
+10.10.10.11
+```
+
+As you can see, the indexes are 1-based.  The syntax when getting or
+setting a `leaf-list` directly by path is similar to the `list` case:  a
+sequence of whitespace-delimited bare values.
+
+```
+$ snabb config get ID /public-ip
+10.10.10.10
+10.10.10.11
+$ snabb config set ID /public-ip "12.12.12.12 13.13.13.13"
+$ snabb config get ID /public-ip
+12.12.12.12
+13.13.13.13
```

Comment:
No because "set" replaces the contents of the array.  You would use "add" to add items